docs/proposals/apb-state-support.md

```diff
@@ -0,0 +1,60 @@
+# Abstract
+
+This proposal will outline a simple and secure method for allowing APB developers to set state during a APB
+action and then be able to access that state during a subsequent APB action.
+
+## Actions
+   - provision
+   - deprovision
+   - update
+   - bind 
+   - unbind
+   
+   
+## Problem Description
+
+APBs are stateless. All state is managed for them by the broker. Currently the broker passes in parameters specified
+during a request to the catalog. It also passes in some additional parameters such as the namespace etc, additionally 
+we also pass in credentials created during the provision. While this is useful, there is no mechanism for an APB to store 
+and access data across actions without exposing it to the end user. We want to avoid APB developers working around this 
+limitation by doing suboptimal things such as storing extra data in the credentials or by creating a ConfigMap in the users 
+namespace as both of these expose the data to the user and as the namespace is controlled by the user, it is naturally 
+not trustworthy or reliable.
+
+
+
+# Proposed Solution
+
+## APB module for handling state
+
+This module will expose an API to the APB developer for setting and getting key value pairs.
+
+```
+
+- name: Save some stuff
+  asb_set_state:
+    service_name: "{{ service_name }}"
+
+```
+
+Under the hood this APB module would take the key value pair, and store it in a
+ConfigMap (or possibly a secret) labelled ```apb:state```. This ConfigMap would live within the 
```

Comment:
I wonder if a name for the config map/secret that conforms to a certain standard might be more appropriate then a label here?

Reply:
I like the idea of following the pattern we established for bind credentials. Create a configmap with name=`POD_NAME` in namespace=`POD_NAMESPACE`

Reply:
No issue with this approach will update 👍